pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3433,31 +3433,16 @@ public String startReplaceSegments(String tableNameWithType, List<String> segmen
   public void endReplaceSegments(String tableNameWithType, String segmentLineageEntryId) {
     try {
       DEFAULT_RETRY_POLICY.attempt(() -> {
```

Comment:
Do we still need this outer retry logic if the logic is retried when trying to update znode?

Reply:
Good point. We still need to keep the top level retry logic to be backward compatible. 

- I have seen the cases where waiting for segments to become online takes ~30minutes (think about the refresh push for TBs of data). 
- Previous implementation was doing the retry for 5 times + wait 10 minutes for each round so we will be successful.
- If we remove the top level, we will fail after 10min waiting time.